linter/helper.go

```diff
@@ -342,3 +343,37 @@ func hasFastlyBoilerPlateMacro(commentText, phrase string) bool {
 	}
 	return false
 }
+
+// According to fastly if a prober is configured with initial < threshold
+// then the prober will be marked as unhealthy at the beginning which is can
+// cause issues at startup.
+func isProbeMakingTheBackendStartAsUnhealthy(prober ast.BackendProbeObject) error {
+	var threshold int
+	var initial int
+	var err error
+	for _, v := range prober.Values {
+		if strings.EqualFold(v.Key.Value, "initial") {
+			// Optimistically cast this to int, if we can do it because the value is a literal
+			// we can apply the check
+			initial, err = strconv.Atoi(v.Value.String())
+			if err != nil {
+				return nil
+			}
+		}
+
+		if strings.EqualFold(v.Key.Value, "threshold") {
+			// Optimistically cast this to int, if we can do it because the value is a literal
+			// we can apply the check
+			threshold, err = strconv.Atoi(v.Value.String())
+			if err != nil {
+				return nil
+			}
+		}
+	}
+
```

Comment:
Do you think we need to care about the case where one of the options is not provided?

Reply:
I thought about this so I wrote the code in such a way that the following happens:

* if both dont appear then both have values 0 and 0 in which case you get no warning
* if initial exists but no threshold then it is also correct because initial value > 0
* if threshold exists but no initial then it is also correct because threshold > 0

Reply:
added a comment in e77ceccb2c4448bb01a0dfb3c1e11ed00d5c6791 to have this as part of the code as well